network/transparent_vlan_endpointclient_linux.go

```diff
@@ -513,15 +513,17 @@ func (client *TransparentVlanEndpointClient) DeleteEndpointsImpl(ep *endpoint, g
 
 	log.Printf("[transparent vlan] There are %d routes remaining after deletion", routesLeft)
 
-	if routesLeft <= numDefaultRoutes {
```

Comment:
Just curious, will commenting this out won't affect the unit test cases we run for it ? https://github.com/Azure/azure-container-networking/blob/master/network/transparent_vlan_endpointclient_linux_test.go#L467

If so, do we need to update the unit test cases as well ?

Reply:
correct updated it

Reply:
good question. these namespaces are created per vlanid. max vlan id support is 4096 and max network namespace that user can create is 27K. so at max we wont go above creating 4K namespace per vm. so should be safe even if we dont remove namespace that we created.

root@aks-nodepool1-36637793-vmss000000:/# cat /proc/sys/user/max_net_namespaces
27653

Reply:
But this does limit the user's ability to go beyond 23k( approximately) even if the vlan is not at max i.e 4k 
I understand that this is safe as we don't have the requirement or a user that has that many namespace, but why are we doing it? Isn't it is safer to delete and make sure it is created when needed, instead of letting it be there.

OR Does this help in reducing the latency since we don't need to create a namespace again ? 
(Do correct me if I am understanding it wrong)

Reply:
so this will be a problem when we parallelize the execution of cni in future.. one doing add and other doing delete in same namespace based on number of default routes